src/NEPTypes.jl

```diff
@@ -1042,33 +1042,39 @@ Returns true/false if the NEP is sparse (if compute_Mder() returns sparse)
     include("nep_transformations.jl")
 
     # structure exploitation for DEP (TODO: document this)
-    function compute_Mlincomb(nep::DEP,λ::Number,V::AbstractVecOrMat,
+    function compute_Mlincomb!(nep::DEP,λ::Number,V::AbstractVecOrMat,
                               a::Vector=ones(eltype(V),size(V,2)))
         n=size(V,1); k=size(V,2);
         Av=get_Av(nep)
         # determine type froom greates type of (eltype probtype and λ)
-        T=promote_type(promote_type(eltype(V),typeof(λ)),eltype(Av[1]),eltype(nep.tauv))
-        if (k>1)
-            broadcast!(*,V,V,transpose(a))
+        TT=promote_type(eltype(V),typeof(λ),eltype(Av[1]),eltype(nep.tauv),eltype(a))
+
+        if (V isa AbstractVector)
+            rmul!(V,a[1])
         else
-            V=V*a[1];
+            broadcast!(*,V,V,transpose(a))
         end
 
-
-        z=zeros(T,n)
+        z=zeros(TT,n)
+        Vw = Vector{TT}(undef, n)
+        AVw = Vector{TT}(undef, n)
         for j=1:length(nep.tauv)
-            w=Array{T,1}(exp(-λ*nep.tauv[j])*(-nep.tauv[j]) .^(0:k-1))
-            if k>1
-                z[:]+=Av[j+1]*(V*w);
-            else
-                z[:]+=Av[j+1]*(V*w[1]);
-            end
+            w=Array{TT,1}(exp(-λ*nep.tauv[j])*(-nep.tauv[j]) .^(0:k-1))
+            mul!(Vw, V, w)
+            mul!(AVw, Av[j+1], Vw)
+            z[:] .+= AVw;
+        end
+
+        if (V isa AbstractVector) || k == 1
+            z-=rmul!(V,λ)
```

Comment:
I would use a fuse subtrastraction ` .-=`, for efficiency and in order to not potentially change the type of `z`. The case `V` vector and matrix need to be handled differently? Or maybe with `V[:]`?